startup/listener.go

```diff
@@ -30,18 +31,20 @@ type Listeners struct {
 
 type ListenersOption func(*Listeners)
 
+// WithListener adds a  listeners. Nil listener is ignored.
 func WithListener(h Listener) ListenersOption {
 	return func(l *Listeners) {
-		if h != nil {
+		if h != nil && !reflect.ValueOf(h).IsNil() {
 			l.listeners = append(l.listeners, h)
 		}
 	}
 }
 
+// WithListeners add multiple listeners. Nil listeners are ignored.
 func WithListeners(h []Listener) ListenersOption {
```

Comment:
Same comment as above, why not `WithListeners(listeners ...Listener)`

Reply:
all fixed